openedx/core/djangoapps/safe_sessions/middleware.py

```diff
@@ -317,13 +317,18 @@ def process_response(self, request, response):
 
         if not _is_cookie_marked_for_deletion(request) and _is_cookie_present(response):
             try:
+                is_from_logout = _is_from_logout(request)
```

Comment:
Can we consider an alternative implementation that doesn't require passing a usage-specific `is_from_logout` boolean parameter around?  I would like to keep this module from not having a baggage of booleans passed around (per Clean Code book etiquette).

One idea: create a context manager for controlling the logging in this middleware.  And then, when is_from_logout is true, set the logging context manager logging level to Error: `logging.basicConfig(level='logging.ERROR')`